partiql-types/src/main/kotlin/org/partiql/value/PartiQLValue.kt

```diff
@@ -65,14 +65,14 @@ public sealed interface ScalarValue<T> : PartiQLValue {
 }
 
 @PartiQLValueExperimental
-public sealed interface CollectionValue<T : PartiQLValue> : PartiQLValue, Sequence<T> {
+public sealed interface CollectionValue<T : PartiQLValue> : PartiQLValue, Iterable<T> {
 
-    public val elements: Sequence<T>?
+    public val elements: Collection<T>?
```

Comment:
I feel as if this should only be `Iterable` as we may not know what is backing the CollectionValue and StructValue. AKA, with the implementations in this PR, we will be materializing all of this data using `toList`.

In any case, it seems like we should have:
```
override fun iterator(): Iterator<T>?
```

Is there a reason we need both `elements` and `iterator`?

Reply:
Iterator gives consumers control over consumption/materialization, but elements is just a simpler interface which does the "toList" for you.

There are many places in the `org.partiql.value.io` which uses `elements` for writing data. Yes, these could all be replace with `.iterator().toList()`, but its less ergonomic imho.

Additionally, the `elements` backing field is nullable whereas `iterator()` will throw an NPE. I don't know if we need both, but elements is like a default way to consume the iterator

Reply:
True, but consumers of PartiQL should be using our APIs for checking nullability. AKA:
```kotlin
val bag = bagValue(someElements)
if (bag.isNull.not()) {
    // Materialize all underlying PartiQL Values
    bag.toList()

    // Loop through elements
    bag.forEach { element -> TODO() }
}
```

Also, notice above that you don't need to use: `.iterator().toList()`.

Reply:
ack, updated